include/wx/gtk/textctrl.h

```diff
@@ -50,6 +50,8 @@ class WXDLLIMPEXP_CORE wxTextCtrl: public wxTextCtrlBase
     virtual int GetLineLength(long lineNo) const wxOVERRIDE;
     virtual wxString GetLineText(long lineNo) const wxOVERRIDE;
     virtual int GetNumberOfLines() const wxOVERRIDE;
+    virtual unsigned GetPhysicalLineCount() const wxOVERRIDE;
+    virtual unsigned GetLogicalLineCount() const wxOVERRIDE;
```

Comment:
Code here: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/TextLayout/Tasks/CountLines.html.

As you can see OSX uses unsigned to have the line count.

Reply:
What about gtk?

Reply:
Well it uses guint and not just gint, right?
Also, I believe you should keep the type for virtual function...

Reply:
> Well it uses guint and not just gint, right?

[gint
gtk_text_iter_get_line (const GtkTextIter *iter)](https://developer.gnome.org/gtk3/stable/GtkTextIter.html#gtk-text-iter-get-line)

Reply:
@catalinr ,
And you missed the second portion of my comment...

Reply:
And you are missing the point...
You chose unsigned int "because OSX uses it", when for a reason just as good you could have chosen int (i.e. "because GTK uses it") and at the same time could have been _consistent_.

Reply:
@catalinr ,
From https://www.tutorialspoint.com/cplusplus/cpp_data_types.htm:

int | 4bytes | -2147483648 to 2147483647
unsigned int | 4bytes | 0 to 4294967295

Do you know a scenario under which the number of lines in the control will be a negative number?
I think the original author of the API was wrong.
I'd rather fix the original API, but I'm afraid this will break the user code.

Reply:
Scenario with more than 2 million lines is just as unrealistic. But I can imagine a scenario, where some SearchLine function would return value -1 as 'not found', like many other wx functions do. And of course, you want to compare and do arithmetics which involves both line count and line index, so you need to have them in same type. Pretty much all other functions use signed value for line number. Should callers to cast their values around every use of these functions, not to get warnings? Changing the de-facto rule for just these two methods would introduce inconsistency in the interface, probably produce lots of signed/unsigned warnings, and bring no gain.

Reply:
@vadz , so what should OSX version return? Also int? Like I said initially Cocoa API chooses unsigned int for some reason. I will check ifif OSX will compile without any warnings anyway.

Reply:
Of course all ports should return the same type. Warnings can be suppressed if really necessary.